tools/src/wyvern/tools/Interpreter.java

```diff
@@ -7,47 +8,45 @@
-import wyvern.stdlib.Globals;
-import wyvern.tools.parsing.Wyvern;
-import wyvern.tools.parsing.transformers.DSLTransformer;
-import wyvern.tools.typedAST.core.expressions.TaggedInfo;
+import wyvern.tools.tests.tagTests.TestUtil;
 import wyvern.tools.typedAST.interfaces.TypedAST;
-import wyvern.tools.typedAST.interfaces.Value;
-import wyvern.tools.util.TreeWriter;
+import wyvern.tools.parsing.coreparser.ParseException;
+import wyvern.tools.typedAST.interfaces.ExpressionAST;
+import wyvern.target.corewyvernIL.expression.Expression;
+import wyvern.target.corewyvernIL.support.TypeContext;
+import wyvern.target.corewyvernIL.support.EvalContext;
+import wyvern.target.corewyvernIL.expression.Value;
+import wyvern.target.corewyvernIL.support.GenContext;
 
 public class Interpreter {
 
 	public static void main(String[] args) {
-		if (args.length != 1) {
+		
+        System.out.println("Running the interpreter.");
+        if(args.length != 1) {
 			System.err.println("usage: wyvern <filename>");
-			System.exit(-1);
-		}
+			System.exit(1);
+        }
 
-		String filename = args[0];
-		Path file = Paths.get(filename);
-		if (!Files.isReadable(file)) {
-			System.err.println("Cannot read file " + filename);
+        String filename = args[0];
+        Path filepath = Paths.get(filename);
+        if(!Files.isReadable(filepath)) {
+            System.err.println("Cannot read file " + filename);
 			System.exit(-1);
-		}
-
-		try {
-			StringReader reader = new StringReader(new String(Files.readAllBytes(file), Charset.forName("UTF-8")) + "\n");
-			TypedAST res = (TypedAST) new Wyvern().parse(reader, filename);
-			res.typecheck(Globals.getStandardEnv(), Optional.empty());
-			// System.out.println("Result 1 = " + res.evaluate(Globals.getStandardEnv()));
-			// res = new DSLTransformer().transform(res); // FIXME: To make to work!
-			Value finalV = res.evaluate(Globals.getStandardEvalEnv());
-			// System.out.println("Result 2 = " + finalV);
-			TreeWriter t = new TreeWriter();
-			finalV.writeArgsToTree(t);
-			System.out.println(t.getResult());
-		} catch (IOException e) {
-			System.err.println("Error reading file " + filename);
-		} catch (CopperParserException e) {
-			System.err.print("Parsing error: ");
-			e.printStackTrace();
-		}
-	}
+        }
 
+        // TODO include the try/catch block for the parse error.
+        String source = TestUtil.readFile(filepath.toAbsolutePath().toString());
+        try {
+            ExpressionAST ast = (ExpressionAST) TestUtil.getNewAST(source);
+            Expression program = ast.generateIL(GenContext.empty());
+            TypeContext ctx = TypeContext.empty();
+            program.typeCheck(ctx);
+            Value v = program.interpret(EvalContext.empty());
+            System.out.println(v.toString());
+        } catch(ParseException e) {
+            System.err.println("Failed to parse file.");
+            e.printStackTrace();
+        }
```

Comment:
This is fine, but even better would be to also catch any ToolError (e.g. a typechecking error) and print the error message on standard error.